protocol/x/prices/keeper/slinky_adapter.go

```diff
@@ -105,3 +105,45 @@ func (k Keeper) GetPrevBlockCPCounter(ctx sdk.Context) (uint64, error) {
 
 	return numMarketPrices, nil
 }
+
+func (k Keeper) GetNumCurrencyPairs(ctx sdk.Context) (uint64, error) {
+	marketPriceStore := k.getMarketPriceStore(ctx)
+
+	var numMarketPrices uint64
+
+	iterator := marketPriceStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		marketPrice := types.MarketPrice{}
+		k.cdc.MustUnmarshal(iterator.Value(), &marketPrice)
+		numMarketPrices++
+	}
+
+	return numMarketPrices, nil
+}
```

Comment:
This is a 1:1 copy of `GetPrevBlockCPCounter`, looking at the `OracleKeeper` interface, it doesn't seem like we need the `GetPrevBlockCPCounter` method anymore?